src/oscar/apps/partner/strategy.py

```diff
@@ -210,14 +210,11 @@ class StockRequired(object):
     def availability_policy(self, product, stockrecord):
         if not stockrecord:
             return Unavailable()
-        if not stockrecord.price:
-            try:
-                if stockrecord.price == None:
-                    return Unavailable()
-            except:
-                return Available()
         if not product.get_product_class().track_stock:
-            return Available()
+            if stockrecord.price == None:
```

Comment:
We check for None in python like this:
```
if stockrecord.price is None:
```
That is because None is a singleton object there exists only 1 instance of NoneType. You can not check for equality using ``is`` with other types!